daisy_workflows/image_import/suse/translate.py

```diff
@@ -26,67 +26,159 @@
 import utils.diskutils as diskutils
 
 
-network = '''
-BOOTPROTO='dhcp'
-STARTMODE='auto'
-DHCLIENT_SET_HOSTNAME='yes'
-'''
+class Package:
+  name = ''
+  gce = False
+  required = False
+
+  def __init__(self, name, gce, required):
+    self.name = name
+    self.gce = gce
+    self.required = required
+
+
+class SuseRelease:
+  flavor = ''
+  major = ''
+  minor = ''
+  packages = None
+  subscriptions = None
+
+  def __init__(self, flavor, major, minor, packages, subscriptions=None):
+    self.flavor = flavor
+    self.major = major
+    self.minor = minor
+    self.packages = packages
+    self.subscriptions = subscriptions
+
+
+cloud_init = Package('cloud-init', gce=False, required=False)
+gcp_sdk = Package('google-cloud-sdk', gce=True, required=False)
+gce_init = Package('google-compute-engine-init', gce=True, required=True)
+gce_oslogin = Package('google-compute-engine-oslogin', gce=True, required=True)
+
+_distros = [
+    SuseRelease(
+        flavor='opensuse',
+        major='15',
+        minor='1',
+        packages=[cloud_init, gcp_sdk, gce_init, gce_oslogin]
+    ),
+    SuseRelease(
+        flavor='sles',
+        major='15',
+        minor='1',
+        packages=[cloud_init, gcp_sdk, gce_init, gce_oslogin],
+        subscriptions=['sle-module-public-cloud/15.1/x86_64']
+    ),
+    SuseRelease(
+        flavor='sles',
+        major='12',
+        minor='5',
+        packages=[cloud_init, gcp_sdk, gce_init, gce_oslogin],
+        subscriptions=['sle-module-public-cloud/12/x86_64']
+    ),
+]
+
+
+def get_distro(g) -> SuseRelease:
+  for d in _distros:
+    if d.flavor == g.gcp_image_distro:
+      if d.major == g.gcp_image_major or d.major == '*':
+        if d.minor == g.gcp_image_minor or d.minor == '*':
+          return d
+  raise AssertionError('Import script not defined for {} {}.{}'.format(
+      g.gcp_image_distro, g.gcp_image_major, g.gcp_image_minor))
+
+
+def install_subscriptions(distro, g):
+  if distro.subscriptions:
+    for subscription in distro.subscriptions:
+      try:
+        g.command(['SUSEConnect', '-p', subscription])
+      except Exception as e:
+        raise ValueError(
+            'Command failed: SUSEConnect -p {}: {}'.format(subscription, e))
+
+
+def install_packages(distro, g, install_gce):
+  g.command(['zypper', 'refresh'])
```

Comment:
You may want to catch errors here as well. In the case there is no repo access or there are errors.

Reply:
done